### PR TITLE
Distro features

### PR DESCRIPTION
**Pull request by angolini**
## Readiness

* [x] Merge (pending reviews)

## Overview

Include how to change (append and remove) and examples for DISTRO_FEATURES, IMAGE_FEATURES, and MACHINE_FEATURES.

any idea on what else to add?

### CODE REVIEW

**Comment by doanac** [comment]
Docs for b44a06326fc69a4fb476b025ade72a9d7c30a34a are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2275/docs/artifacts/html/index.html

File: source/reference-manual/linux/linux-extending.rst [hunk 1]
```diff
@@ -105,6 +106,74 @@ Update the following variables to reflect the details from the package you wish
...
 
+Using FEATURES to configure LmP
+--------------------------------------
+
+DISTRO_FEATURES is a list of configurations from a distro that reflects how
+some packages are built or installed. There is a list of `Yocto Project distro features`_
+supported. However, the list can be expanded by other meta layers.
+
+For example, the distro feature ``systemd`` or ``wayland`` are used to define the
+list of packages to be installed, and to configure some packages build. The
```
Comment (kprosise) [inline source/reference-manual/linux/linux-extending.rst:117]:
```suggestion
list of packages to be installed, and to configure how some packages build. The
```

File: source/reference-manual/linux/linux-extending.rst [hunk 2]
```diff
@@ -105,6 +106,74 @@ Update the following variables to reflect the details from the package you wish
...
+  DISTRO_FEATURES="acl argp bluetooth ext2 ipv4 ipv6 largefile usbgadget usbhost wifi xattr zeroconf pci vfat modsign efi security tpm integrity seccomp pam usrmerge virtualization ptest
+  alsa sota systemd"
+
+The log is generated using ``DISTRO="lmp"``. The ``DISTRO_FEATURES`` changed with
+seven operations and only one of them is for an override (``tegra``).
+The log also shows the file path and line for each operation.
+
+The line starting with ``DISTRO_FEATURES=`` show the variable value.
+
+Yocto Project provides also the ``IMAGE_FEATURES`` and the ``MACHINE_FEATURES``,
```
Comment (kprosise) [inline source/reference-manual/linux/linux-extending.rst:159]:
```suggestion
The Yocto Project also provides ``IMAGE_FEATURES`` and ``MACHINE_FEATURES``,
```

File: source/reference-manual/linux/linux-extending.rst [hunk 3]
```diff
@@ -105,6 +106,74 @@ Update the following variables to reflect the details from the package you wish
...
+a list of features for the image and to describe the machine. There is a list of
+`Yocto Project image features`_ and `Yocto Project machine features`_ supported
+by the project.
+
+The LmP uses the ``MACHINE_FEATURES`` from a machine to define if a package is included.
+For example, the OP-Tee package is only included in an image if the target machine
+includes the feature ``optee`` in the ``MACHINE_FEATURE``.
```
Comment (kprosise) [inline source/reference-manual/linux/linux-extending.rst:166]:
```suggestion
includes the feature ``optee`` in ``MACHINE_FEATURE``.
```

**Review by kprosise — APPROVED** [review]
Left some suggestions, but it is looking good!

**Comment by kprosise** [comment]
At the moment, I don't know what else that may be appropriate to add, but I will let you know if anything comes to mind later.

**Comment by doanac** [comment]
Docs for 35fca6bb06f9a2a2da98c6b58cfbd0b70b685211 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2276/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for dfa4ecbd6e33a90350b9e7a9a8c8d29d75845e36 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2277/docs/artifacts/html/index.html

**Comment by angolini** [comment]
> @angolini shall I go ahead and merge?
@kprosise yes, please!

**Pull request merged**